### DISCUSSION TITLE
How I can prevent duplicate log lines between different sinks?

### DISCUSSION THREAD

**Discussion by afshinpir** [gabime/spdlog#2042] [Q&A]
Hi

I'm trying to use 3 sinks like this:
```
    try
    {
        auto console_sink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
        console_sink->set_level(spdlog::level::info);

        auto console_err_sink = std::make_shared<spdlog::sinks::stderr_color_sink_mt>();
        console_err_sink->set_level(spdlog::level::error);

        auto file_sink = std::make_shared<spdlog::sinks::basic_file_sink_mt>("logs/multisink.txt", true);
        file_sink->set_level(spdlog::level::trace);

        spdlog::sinks_init_list sink_list = { file_sink, console_sink, console_err_sink };

        spdlog::logger logger("multi_sink", sink_list.begin(), sink_list.end());
        logger.set_level(spdlog::level::trace);
    }
    catch (const spdlog::spdlog_ex& ex)
    {
        std::cout << "Log initialization failed: " << ex.what() << std::endl;
    }
```
and I like all logs will be available in file, error goes to `stderr` and info goes to `stdout`. But since `stderr` and `stdout` is display by default, I will see 2 similar line of logs from 2 sinks in display. I can solve this problem by using 2 loggers, But I wonder if it is possible to use 1 logger and achieve what I need or not.
Should I create custom sink by inheriting from `spdlog::sinks::stdout_color_sink_mt` and limiting max log level?

**Comment by tt4g** [comment]
> Should I create custom sink by inheriting from spdlog::sinks::stdout_color_sink_mt and limiting max log level?

Yes. but I recommended to inherits `spdlog::sinks::base_sink` and implement proxy-pattern.

See this comment: https://github.com/gabime/spdlog/issues/1363#issuecomment-567068416
